### PR TITLE
Update lint.yml

### PR DESCRIPTION
**Pull request by okwme**
opening in place of #1509 to make gaia consistent with cosmos sdk

### CODE REVIEW

File: .golangci.yml [hunk 1]
```diff
@@ -58,9 +50,11 @@ issues:
     - text: "ST1016:"
       linters:
         - stylecheck
-    - text: "SA1019: codec.LegacyAmino is deprecated"
+    - path: "migrations"
+      text: "SA1019:"
```
Comment (tokamak-git) [inline .golangci.yml:55]: @okwme not quiet sure  i understand this bit. is the `text` field used as a regex? if so we are changing it?

Comment (okwme) [reply .golangci.yml:55]: would actually need to ask @marbar3778 or @alexanderbez i just copied from SDK's config

File: .github/workflows/lint.yml [hunk 2]
```diff
@@ -1,25 +1,41 @@
...
-    name: Run golangci-lint
+    name: golangci-lint
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v3
       - uses: actions/setup-go@v3
         with:
-          go-version: 1.17
+          go-version: 1.18
       - uses: technote-space/get-diff-action@v6.0.1
```
Comment (faddat) [inline .github/workflows/lint.yml:18]: This right here is the problem

File: .github/workflows/lint.yml [hunk 3]
```diff
@@ -1,25 +1,41 @@
...
+        id: git_diff
         with:
           PATTERNS: |
             **/**.go
             go.mod
             go.sum
-      - name: Run golangci-lint
-        run: make lint
+      - name: Get data from Go build cache
```
Comment (faddat) [inline .github/workflows/lint.yml]: this right here is another problem.

File: .golangci.yml (hunk 1 above)
Comment (faddat) [reply .golangci.yml:55]: The sdk's config is why this isn't firing the linter:

https://github.com/cosmos/cosmos-sdk/pull/11453

Comment (tac0turtle) [reply .golangci.yml:55]: I don't think this is needed. @faddat is right in why it's not firing

Comment (okwme) [reply .golangci.yml:55]: @marbar3778 can you make changes or specific suggestions of what is needed to change before merge?

File: .github/workflows/lint.yml [hunk 4]
```diff
@@ -1,25 +1,41 @@
...
+        # if: env.GIT_DIFF
+        if: ${{ false }}
+        uses: actions/cache@v3
+        with:
+          path: |
+            ~/go/pkg/mod
+            ~/.cache/golangci-lint
+            ~/.cache/go-build
+          key: ${{ runner.os }}-go-build-${{ hashFiles('**/go.sum') }}
```
Comment (tac0turtle) [inline .github/workflows/lint.yml]:
```suggestion
```

Comment (tac0turtle) [reply .github/workflows/lint.yml]: this isn't needed as golangci=lint caches on its own. we need to clean up in the sdk

File: .github/workflows/lint.yml (hunk 2 above)
Comment (tac0turtle) [reply .github/workflows/lint.yml:18]: this is fine, on gaia is not super needed, but doesn't hurt anyone

**Review by tac0turtle — APPROVED** [review]

**Pull request merged**